packages/@sanity/types/src/schema/define.ts

```diff
@@ -170,7 +170,7 @@ import {FieldDefinitionBase, IntrinsicTypeName} from './definition'
  * @beta
  */
 export function defineType<
-  TType extends string | IntrinsicTypeName, // IntrinsicTypeName here improves autocompletion in _some_ IDEs (not VS Code atm)
+  TType extends IntrinsicTypeName | (string & {}), 
```

Comment:
Awesome that this works, but this seems like is a real head-scratcher. Could you add a comment indicating why it is there/why it is necessary for future maintainers? (without it it could possibly be deleted under the guise of "this is redundant / what does string & {} even mean)

Reply:
I'll add a comment but if I'm being completely honest with you I also don't know why it works 😅